src/olympia/blocklist/tasks.py

```diff
@@ -143,16 +147,35 @@ def import_block_from_blocklist(record):
 
 
 @task
-def upload_filter_to_kinto(generation_time):
+def upload_filter_to_kinto(generation_time, is_base=True, upload_stash=False):
     server = KintoServer(
         KINTO_BUCKET, KINTO_COLLECTION_MLBF, kinto_sign_off_needed=False)
+    mlbf = MLBF(generation_time)
+    if is_base:
+        # clear the collection for the base - we want to be the only filter
+        server.delete_all_records()
+    # Deal with possible stashes first
```

Comment:
I'm a bit confused by the implementation.

I am expecting a fixed MLBF base plus some (none, one or more) stashes. And maybe one continuously updating MLBF, if we want to disable stashes.

The current implementation always published a new MLBF. So the collection may look like.

```
{ stash_time: 1, ... }, (or maybe not, if upload_stash=False when is_base=True)
{ stash_time: 2, ... },
{ stash_time: 3, ... },
{ stash_time: 4, ... },
{ generation_time: 1, attachment_type: "bloomfilter-base", attachment: ... }, 
{ generation_time: 2, attachment_type: "bloomfilter-update", attachment: ... }, 
{ generation_time: 3, attachment_type: "bloomfilter-update", attachment: ... }, 
{ generation_time: 4, attachment_type: "bloomfilter-update", attachment: ... }, 
```

This has too many bloomfilter-update attachments. I would expect something like this:

```
{ stash_time: 2, ... },
{ stash_time: 3, ... },
{ stash_time: 4, ... },
{ generation_time: 1, attachment_type: "bloomfilter-base", attachment: ... }, 
{ generation_time: 4, attachment_type: "bloomfilter-full", attachment: ... },
```
Note: "full" instead of "update" to better reflect its meaning.

Reply:
The pre-stash client side implementation mentioned previously (if I understood it correctly) was to look at all records that had `generation_time` and just download the attachment from the one with the latest `generation_time`.  In that implementation it wouldn't matter how many MLBF attachments there were in the collection.  Has that been changed?

We could keep track of the `bloomfilter-update`/`bloomfilter-full` records and update them I suppose, I just didn't see a need to.  (I'll do it as a follow-up issue - this pr has enough moving parts as it is)

Reply:
> The pre-stash client side implementation mentioned previously (if I understood it correctly) was to look at all records that had `generation_time` and just download the attachment from the one with the latest `generation_time`. In that implementation it wouldn't matter how many MLBF attachments there were in the collection. Has that been changed?

From the client's perspective it doesn't matter. Avoiding duplication could indeed be done in a follow-up.

I still suggest to go with the "-full" suffix right away rather than "-update", because "-full" more accurately represents the meaning of the MLBF.

Reply:
https://github.com/mozilla/addons-server/pull/13986/commits/f5d3fb70189c0dcdfe2704ef26516d9f8c7eb8e3 and #14013 for follow-up